# **Healthcare Insurance Cost Analysis**

## Objectives

The objective of this notebook is to analyse a healthcare insurance dataset and understand how personal, lifestyle and geographic factors influence medical insurance charges.

This notebook will:
* Load and inspect the healthcare insurance dataset
* Check the dataset for missing values, duplicates and incorrect data types
* Clean and prepare the data for analysis
* Explore relationships between insurance charges and variables such as age, BMI, smoking status, sex, children and region
* Create visualisations to support the analysis
* Summarise the key findings, limitations and possible next steps

## Inputs

The input for this notebook is the `insurance.csv` dataset stored in the `data` folder.

The dataset contains the following columns:

* `age`: age of the insurance customer
* `sex`: gender of the customer
* `bmi`: body mass index
* `children`: number of children/dependants covered by insurance
* `smoker`: whether the customer is a smoker
* `region`: customer region
* `charges`: individual medical insurance charges

## Outputs

The outputs of this notebook will include:

* An inspected and cleaned version of the insurance dataset, if cleaning is required
* Summary statistics for the dataset
* Exploratory data analysis tables
* Visualisations showing patterns in insurance charges
* Written insights explaining which factors appear to influence insurance charges
* A final conclusion and suggested next steps

## Additional Comments

This notebook will be developed step by step, starting with loading and understanding the dataset before moving on to cleaning, exploratory data analysis, visualisations and conclusions.



---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'/Users/sahraosman/Documents/vscode-projects/Healthcare_Insurance/jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chdir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'/Users/sahraosman/Documents/vscode-projects/Healthcare_Insurance'

---

# Section 1: Load and Inspect the Dataset

In this section, the required Python libraries are imported for data analysis and visualisation. Pandas is then used to load the healthcare insurance dataset and inspect its initial structure.

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [5]:
df = pd.read_csv('data/insurance.csv')
df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


The first five rows show that the dataset has loaded successfully. It contains customer information including age, sex, BMI, number of children, smoking status, region and insurance charges.

---

# Section 2: Initial Data Inspection

In this section, the dataset will be inspected to understand its size, column names, data types and summary statistics.

In [6]:
df.shape

(1338, 7)

The dataset contains 1,338 rows and 7 columns.

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


The dataset has no missing values shown in the non-null count, as each column contains 1,338 non-null entries. There are four numerical columns: `age`, `bmi`, `children` and `charges`, and three categorical columns: `sex`, `smoker` and `region`.

In [8]:
df.describe()

,age,bmi,children,charges
count,1338.000000,1338.000000,1338.000000,1338.000000
mean,39.207025,30.663397,1.094918,13270.422265
std,14.049960,6.098187,1.205493,12110.011237
min,18.000000,15.960000,0.000000,1121.873900
25%,27.000000,26.296250,0.000000,4740.287150
50%,39.000000,30.400000,1.000000,9382.033000
75%,51.000000,34.693750,2.000000,16639.912515
max,64.000000,53.130000,5.000000,63770.428010


The summary statistics show that customer ages range from 18 to 64, with an average age of approximately 39. The average BMI is approximately 30.66, and the number of children ranges from 0 to 5. Insurance charges vary widely, from around 1,121.87 to 63,770.43, which makes it important to explore which customer characteristics may be associated with higher or lower charges.

In [9]:
df.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

The column names are clear, consistent and lowercase. They do not contain spaces, so no column renaming is required at this stage.

---

# Section 3: Data Quality Checks

In this section, the dataset will be checked for duplicates, missing values, incorrect data types and any other issues that may require cleaning before analysis.

In [10]:
df.duplicated().sum()

1

This shows that there is 1 duplicate row in the dataset. The duplicate row will be removed to avoid counting the same record more than once in the analysis.

In [11]:
df.drop_duplicates(inplace=True)

This removes the duplicate row from the dataset, ensuring that each customer is only counted once in the analysis.

In [12]:
df.duplicated().sum()

0

This shows there are no duplicate rows in the dataset after removing the duplicate, as the `duplicated().sum()` method returns 0. Therefore, no further duplicate removal is required.

In [13]:
df.shape

(1337, 7)

After removing the duplicate row, the dataset now contains 1,337 rows and 7 columns.

In [14]:
df.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

This shows there are no missing values in the dataset, as the `isnull().sum()` method returns a Series of all zero values. Therefore, no missing value imputation is required.

In [15]:
df.dtypes

age           int64
sex          object
bmi         float64
children      int64
smoker       object
region       object
charges     float64
dtype: object

This shows the different data types in the dataset. The `age`, `bmi`, `children` and `charges` columns are numeric, while the `sex`, `smoker` and `region` columns are object types. These data types are appropriate for the initial analysis, so no data type conversion is required at this stage.

---

# Section 4: Exploratory Data Analysis

This section explores the relationships between insurance charges and the other variables in the dataset. Summary statistics and grouped analysis will be used to identify early patterns before creating visualisations.

## Target Variable: Charges

In [16]:
df['charges'].describe().round(2)

count     1337.00
mean     13279.12
std      12110.36
min       1121.87
25%       4746.34
50%       9386.16
75%      16657.72
max      63770.43
Name: charges, dtype: float64

This shows the characteristics of the `charges` column, which is the target variable for this analysis. The mean insurance charge is approximately 13,279.12, with a standard deviation of 12,110.36. The minimum charge is 1,121.87, while the maximum charge is 63,770.43. The median charge is 9,386.16, indicating that half of the customers have charges below this amount and half have charges above it. The mean is higher than the median, which suggests that the distribution of insurance charges is right-skewed, with a few customers having very high charges.

## Categorical and Count Variable Distributions

In [17]:
df['sex'].value_counts()

sex
male      675
female    662
Name: count, dtype: int64

There are 662 female customers and 675 male customers in the dataset, indicating a fairly balanced gender distribution. This means both groups are well represented, making it reasonable to compare insurance charges by sex later in the analysis.

In [18]:
df['smoker'].value_counts()

smoker
no     1063
yes     274
Name: count, dtype: int64

There are more non-smokers (1,063) than smokers (274) in the dataset. This shows that smoking status is imbalanced, with non-smokers making up most of the records. This should be considered when comparing insurance charges by smoking status later in the analysis.

In [19]:
df['region'].value_counts()

region
southeast    364
southwest    325
northwest    324
northeast    324
Name: count, dtype: int64

The regional distribution is fairly balanced. The southeast has the highest number of customers with 364 records, while the other regions have similar counts: 325 from the southwest, 324 from the northwest, and 324 from the northeast. This means all regions are well represented, making it reasonable to compare insurance charges by region later in the analysis.

In [20]:
df['children'].value_counts().sort_index()

children
0    573
1    324
2    240
3    157
4     25
5     18
Name: count, dtype: int64

This shows the number of children covered by insurance for each customer. The majority of customers (573) have no children, while 324 customers have one child, 240 have two children, 157 have three children, 25 have four children, and 18 have five children. This shows that customers with no children are the largest group, while customers with four or five children are much less common.

## Numerical Variable Distributions

In [23]:
df[['age', 'bmi']].describe().round(2)

,age,bmi
count,1337.00,1337.00
mean,39.22,30.66
std,14.04,6.10
min,18.00,15.96
25%,27.00,26.29
50%,39.00,30.40
75%,51.00,34.70
max,64.00,53.13


For age and BMI, the summary statistics show that the average age of customers is approximately 39 years, with a minimum age of 18 and a maximum age of 64. The average BMI is approximately 30.66, with a minimum BMI of 15.96 and a maximum BMI of 53.13. The median age is 39, which is very close to the mean, suggesting the age distribution is fairly balanced. The median BMI is 30.40, which is also close to the mean BMI of 30.66. This suggests that the BMI distribution is not heavily skewed, although the maximum BMI of 53.13 shows that some customers have much higher BMI values.

## Average Insurance Charges by Group

In [24]:
df.groupby('smoker')['charges'].mean().round(2)

smoker
no      8440.66
yes    32050.23
Name: charges, dtype: float64

This shows that non-smokers have an average insurance charge of approximately 8,440.66, while smokers have a much higher average charge of approximately 32,050.23. This suggests that smoking status appears to have a strong relationship with insurance charges, with smokers having significantly higher average costs than non-smokers.

In [25]:
df.groupby('sex')['charges'].mean().round(2)

sex
female    12569.58
male      13975.00
Name: charges, dtype: float64

This shows that females have an average insurance charge of approximately 12,569.58, while males have a slightly higher average charge of approximately 13,975.00. This suggests that there is a small difference in average insurance charges by sex, with males having slightly higher average costs than females.

In [26]:
df.groupby('children')['charges'].mean().round(2)

children
0    12384.70
1    12731.17
2    15073.56
3    15355.32
4    13850.66
5     8786.04
Name: charges, dtype: float64

Customers with no children have an average insurance charge of approximately 12,384.70, while customers with one child have a slightly higher average charge of approximately 12,731.17. Customers with two and three children have higher average charges, at approximately 15,073.56 and 15,355.32. Customers with four children have an average charge of approximately 13,850.66, while customers with five children have the lowest average charge of approximately 8,786.04.

This suggests that the number of children does not show a simple pattern where charges increase steadily as the number of children increases. The lower average charge for customers with five children may be affected by the small sample size, as there are only 18 customers in that group. Therefore, this result should be interpreted with caution.

In [27]:
df.groupby('region')['charges'].mean().round(2)

region
northeast    13406.38
northwest    12450.84
southeast    14735.41
southwest    12346.94
Name: charges, dtype: float64

The region with the highest average insurance charges is the southeast, with an average charge of approximately 14,735.41. The region with the lowest average insurance charges is the southwest, with an average charge of approximately 12,346.94.

The northwest and southwest regions have very similar average charges, while the northeast is slightly higher at approximately 13,406.38. This suggests that there may be some regional differences in insurance charges, especially between the southeast and the other regions. However, the regional differences are much smaller than the difference seen between smokers and non-smokers, so region appears to have a weaker relationship with charges than smoking status.

## Section 4 Summary

The exploratory analysis shows that smoking status appears to have the strongest relationship with insurance charges so far, as smokers have much higher average charges than non-smokers. Sex, region and number of children show some differences in average charges, but these differences are smaller and less consistent than the difference between smokers and non-smokers.

The number of children does not show a simple pattern where charges increase steadily, and the result for customers with five children should be interpreted carefully due to the small sample size. The next section will use visualisations to explore these patterns more clearly.